src/main/java/mate/jdbc/dao/DriverDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Driver;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class DriverDaoImpl implements DriverDao {
+    @Override
+    public Driver create(Driver driver) {
+        String createDriverRequest = "INSERT INTO drivers(name, licenseNumber) values(?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createDriverStatement =
+                        connection
+                                .prepareStatement(createDriverRequest,
+                                        Statement.RETURN_GENERATED_KEYS)) {
+            createDriverStatement.setString(1, driver.getName());
+            createDriverStatement.setString(2, driver.getLicenseNumber());
+            createDriverStatement.executeUpdate();
+            ResultSet generatedKeys = createDriverStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                driver.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't insert driver to DB.", e);
+        }
+        return driver;
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        String getDriverRequest = "SELECT * FROM drivers WHERE is_deleted = false AND id = ?;";
+        Driver driver = new Driver();
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriverStatement =
+                        connection.prepareStatement(getDriverRequest)) {
+            getDriverStatement.setLong(1, id);
+            getDriverStatement.executeQuery();
+            ResultSet resultSet = getDriverStatement.getResultSet();
+            if (resultSet.next()) {
+                String name = resultSet.getString("name");
+                String licenseNumber = resultSet.getString("licenseNumber");
+                driver.setName(name);
+                driver.setLicenseNumber(licenseNumber);
+                driver.setId(id);
+                return Optional.of(driver);

```

Comment:
we can move duplicated logic to separate method

Reply:
done